File: data-tool/src/main/java/com/nilecochen/fitnessapp/datatool/entities/Workout.java
package com.nilecochen.fitnessapp.datatool.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "workouts")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Workout {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "workout_date", nullable = false)
    private LocalDate workoutDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "workout_kind")
    private String workoutKind;

    @Column(columnDefinition = "vector(1024)")
    private String embeddings;

    @Column(columnDefinition = "TEXT")
    private String workoutText;

    // Relationships
    @OneToMany(mappedBy = "workout", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<WorkoutExercise> workoutExercises = new HashSet<>();
}

File: admin-dash/Admin-Dash/Models/Workout.cs
using System;
using System.Collections.Generic;

namespace Admin_Dash.Models;

public partial class Workout
{
    public long Id { get; set; }

    public DateOnly WorkoutDate { get; set; }

    public long UserId { get; set; }

    public string? WorkoutKind { get; set; }

    public string? WorkoutText { get; set; }

    public string? Workouttext { get; set; }

    public virtual User User { get; set; } = null!;

    public virtual ICollection<WorkoutExercise> WorkoutExercises { get; set; } = new List<WorkoutExercise>();
}
